src/adapters/chineseloc/adapter.js

```diff
@@ -1,127 +1,164 @@
 /* eslint-disable no-unused-vars */
 import BaseAdapter from '@/adapters/base-adapter'
-import ChineseSource from '@/adapters/chineseloc/chinese-source/chinese-source.js'
 import { ChineseLanguageModel, Lemma, Lexeme, Homonym, Feature, Definition } from 'alpheios-data-models'
+import {
+  MessagingService, WindowIframeDestination as Destination, CedictDestinationConfig as CedictConfig,
+  CedictCharacterForms, RequestMessage
+} from 'alpheios-lexis-cs'
+
+const msgServiceName = 'AdaptersLexisService'
 
 class AlpheiosChineseLocAdapter extends BaseAdapter {
   constructor (config = {}) {
     super()
     this.config = config
+
+    /*
+    AlpheiosChineseLocAdapter is created every time when a new lexical request for Chinese data comes in.
+    We do not want to create a new instance of a messaging service with that. Thus, we'll use a single
+    instance of the service that will be created once and reused across consecutive constructor invocations.
+     */
+    if (!MessagingService.hasService(msgServiceName)) {
+      MessagingService.createService(msgServiceName, new Destination(CedictConfig))
+    }
+    this._messagingService = MessagingService.getService(msgServiceName)
   }
 
   get languageID () { return ChineseLanguageModel.languageID }
 
-  fetchChineseData (targetWord, checkContextForward) {
-    ChineseSource.collectData()
-
-    return ChineseSource.lookupChinese(targetWord, checkContextForward)
+  // region Pinyin formatting functions
+  // TODO: Do these functions really belong in here?
+  static isVowel (aLetter) {
+    return ['a', 'e', 'i', 'o', 'u'].includes(aLetter)
   }
 
-  getHomonym (targetWord, checkContextForward) {
-    try {
-      const res = this.fetchChineseData(targetWord, checkContextForward)
-      if (res) {
-        let homonym = this.transformData(res, targetWord)
+  static formatPinyin (aPinyin) {

```

Comment:
May be it is better to place it to ChineseLanguageModel in `data-models`?
It is chinese specific and I believe that it is not a job of the adapter to know how format specific language things?

Reply:
I think it is a good idea. @balmas, what do you think?

Reply:
yes I agree. Not sure if there is a generic need for something like that, so it would be a method only on the Chinese Language Model and not the base model but that's ok.

Reply:
Moved the `formatPinyin` to `data-modules`.